src/internal/ppsdb/ppsdb.go

```diff
@@ -66,37 +104,44 @@ func Pipelines(db *pachsql.DB, listener col.PostgresListener) col.PostgresCollec
 		pipelinesIndexes,
 		col.WithKeyGen(func(key interface{}) (string, error) {
 			if commit, ok := key.(*pfs.Commit); ok {
-				if commit.Branch.Repo.Type != pfs.SpecRepoType {
-					return "", errors.Errorf("commit %s is not from a spec repo", commit)
-				}
-				return fmt.Sprintf("%s@%s", commit.Branch.Repo.Name, commit.ID), nil
+				return pipelineCommitKey(commit)
 			}
 			return "", errors.New("must provide a spec commit")
 		}),
 		col.WithKeyCheck(func(key string) error {
-			_, _, err := ParsePipelineKey(key)
+			_, _, _, err := ParsePipelineKey(key)
 			return err
 		}),
 	)
 }
 
+func JobsPipelineKey(p *pps.Pipeline) string {
+	if projectName := p.Project.GetName(); projectName != "" {
+		return fmt.Sprintf("%s/%s", projectName, p.Name)
+	}
+	return p.Name
```

Comment:
We can’t expect a project name yet, in part because the migration story is not complete, and the migration story is not complete because it depends on full support for projects. This code is in place to break the circular dependency and enable project support to be merged in piecewise as separate stories.